bridge-token-factory/src/lib.rs

```diff
@@ -342,6 +434,212 @@ impl BridgeTokenFactory {
+                return 0;
+            }
+        }
+    }
+
+    //token here should be ethereum address
+    #[access_control_any(roles(Role::FeeSetter))]
+    pub fn set_deposit_fee_bound(
+        &mut self,
+        token: &String,
+        upper_bound: u128,
+        lower_bound: u128,
+    ) -> DepositTokenBounds {
+        let _ = validate_eth_address(token.clone());
+        self.deposit_fee_bound.insert(
+            token,
+            &DepositTokenBounds {
+                lower_bound,
+                upper_bound,
+            },
+        );
+        DepositTokenBounds {
+            lower_bound,
+            upper_bound,
+        }
+    }
+
+    //this should be added as per: 10% -> 0.1 = 0.1*10^6
+    #[access_control_any(roles(Role::FeeSetter))]
+    pub fn set_deposit_fee_percentage(
+        &mut self,
+        token: &String,
+        eth_to_near: u128,
+        eth_to_aurora: u128,
+    ) -> DepositFeePercentage {
+        let _ = validate_eth_address(token.clone());
+        self.deposit_fee_percentage.insert(
+            token,
+            &DepositFeePercentage {
+                eth_to_near,
+                eth_to_aurora,
+            },
+        );
+        DepositFeePercentage {
+            eth_to_near,
+            eth_to_aurora,
+        }
+    }
+
+    // sets fee bounds for withdraw tokens, default bound: None
+    #[access_control_any(roles(Role::FeeSetter))]
+    pub fn set_withdraw_fee_bound(
+        &mut self,
+        token: &String,
+        upper_bound: u128,
+        lower_bound: u128,
+    ) -> WithdrawTokenBounds {
+        let _ = validate_eth_address(token.clone());
+        self.withdraw_fee_bound.insert(
+            token,
+            &WithdrawTokenBounds {
+                lower_bound,
+                upper_bound,
+            },
+        );
+        WithdrawTokenBounds {
+            lower_bound,
+            upper_bound,
+        }
+    }
+
+    //this should be added as per: 10% -> 0.1 = 0.1*10^6
+    #[access_control_any(roles(Role::FeeSetter))]
+    pub fn set_withdraw_fee_percentage(
+        &mut self,
+        token: &String,
+        near_to_eth: u128,
+        aurora_to_eth: u128,
+    ) -> WithdrawFeePercentage {
+        let _ = validate_eth_address(token.clone());
+        self.withdraw_fee_percentage.insert(
+            token,
+            &WithdrawFeePercentage {
+                near_to_eth,
+                aurora_to_eth,
+            },
+        );
+        WithdrawFeePercentage {
+            near_to_eth,
+            aurora_to_eth,
+        }
+    }
+
+    /*
+    Sets fee percentage for withdraw per silo per token
+    @param: fee_percent:- it should be with 6 decimals, eg: 10% = 0.1*10^6
+            silo_address:- should be an aurora account
+     */
+    #[access_control_any(roles(Role::FeeSetter))]
+    pub fn set_withdraw_fee_percentage_for_token_per_silo(
+        &mut self,
+        silo_address: &AccountId,
+        token: &String,
+        fee_percent: u128,
```

Comment:
Please use `U128` instead of `u128` for json input/output

Reply:
fixed in 6a7df359a29f6105fae2cccc36ad5277fb747102